src/engine/BilinearConstraint.h

```diff
@@ -0,0 +1,104 @@
+/*********************                                                        */
+/*! \file BilinearConstraint.h
+** \verbatim
+** Top contributors (to current version):
+**   Andrew Wu
+** This file is part of the Marabou project.
+** Copyright (c) 2017-2019 by the authors listed in the file AUTHORS
+** in the top-level source directory) and their institutional affiliations.
+** All rights reserved. See the file COPYING in the top-level source
+** directory for licensing information.\endverbatim
+**
+** [[ Add lengthier description here ]]
+**/
+
+#ifndef __BilinearConstraint_h__
+#define __BilinearConstraint_h__
+
+#include "List.h"
+#include "Map.h"
+#include "NonlinearConstraint.h"
+
+#include <cmath>
+
+class BilinearConstraint : public NonlinearConstraint
+{
+public:
+    BilinearConstraint( unsigned b1, unsigned b2, unsigned f );
+    BilinearConstraint( const String &serializedBilinear );
+
+    /*
+      Get the type of this constraint.
+    */
+    NonlinearFunctionType getType() const override;
+
+    /*
+      Return a clone of the constraint.
+    */
+    NonlinearConstraint *duplicateConstraint() const override;
+
+    /*
+      Restore the state of this constraint from the given one.
+    */
+    void restoreState( const NonlinearConstraint *state ) override;
+
+    /*
+      Register/unregister the constraint with a talbeau.
+    */
+    void registerAsWatcher( ITableau *tableau ) override;
+    void unregisterAsWatcher( ITableau *tableau ) override;
+
+    /*
+      These callbacks are invoked when a watched variable's value
+      changes, or when its bounds change.
+    */
+    void notifyLowerBound( unsigned variable, double bound ) override;
+    void notifyUpperBound( unsigned variable, double bound ) override;
+
+    /*
+      Returns true iff the variable participates in this nonlinear constraint
+    */
+    bool participatingVariable( unsigned variable ) const override;
+
+    /*
+      Get the list of variables participating in this constraint.
+    */
+    List<unsigned> getParticipatingVariables() const override;
+
+    /*
+      Preprocessing related functions, to inform that a variable has
+      been eliminated completely because it was fixed to some value,
+      or thisat a variable's index has changed (e.g., x4 is now called
+      x2). constraintObsolete() returns true iff and the constraint
```

Comment:
return not true but false?

Reply:
For now we do not support eliminate variables from the bilinear constraints.